envbuilder.go

```diff
@@ -182,6 +135,40 @@ func Run(ctx context.Context, opts options.Options) error {
 			opts.Logger(log.LevelError, "Failed to clone repository: %s", fallbackErr.Error())
 			opts.Logger(log.LevelError, "Falling back to the default image...")
 		}
+
+		// The repo wasn't cloned (i.e. may not be up-to-date with remote), so
+		// we need to clone it to get the right build context. This is only
+		// necessary when the repo is in remote build mode. If the repo is in
+		// local build mode, the build context is already available on the host
+		// filesystem.
+		//
+		// Skipping clone here if we believe workspace repo matches remote repo
+		// is a performance optimization.
```

Comment:
is this a TODO?

Reply:
This was my attempt at explaining why we have the if in relation to the previous clone of the users workspace repo. Not a todo but wording can perhaps be improved. Or we can ditch it and always clone.

Reply:
Does always cloning make the implementation simpler for now?

Reply:
Mostly it's just removal of the one if logic, but less things to go wrong so I suppose it is, yeah. Let's go for simplicity.